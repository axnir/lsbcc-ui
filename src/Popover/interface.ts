export interface PopoverProps {
  /**
   * @description      卡片标题
   */
  title: string;
  /**
   * @description      卡片内容
   */
  content: React.ReactNode;
  /**
   * @description      卡片位置
   */
  placement?: 'top' | 'right' | 'bottom' | 'left';
}
