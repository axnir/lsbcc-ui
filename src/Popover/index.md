---
title: Popover 气泡卡片
nav:
  path: /popover
  title: 组件
  order: 2
group:
  title: 基础组件
  order: 1
---

# Popover 气泡卡片

鼠标移入元素，弹出气泡式的卡片浮层。

## 代码演示

### 基本用法

<code src="./demo/index.tsx"></code>

<API src="./popover.tsx"></API>
