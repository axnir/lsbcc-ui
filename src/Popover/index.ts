import Popover from './popover';

export default Popover;

export * from './interface';
