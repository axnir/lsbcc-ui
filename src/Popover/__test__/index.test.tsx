import React from 'react';
import { render } from '@testing-library/react';
import Popover from '../popover';

const title = 'Title';
const content = (
  <div>
    <p>Hover me</p>
    <p>Hover me</p>
  </div>
)

describe('<Popover />', () => {
  test('should render default', () => {
    const { container } = render(
      <Popover title={title} content={content}>
        <button>left</button>
      </Popover>
    );
    expect(container).toMatchSnapshot();
  });

  test('should render popover with placement', () => {
    const placements: any[] = ['top', 'right', 'bottom', 'left'];

    const { getByText } = render(
      <>
        {placements.map(t => (
          <Popover key={t} placement={t} title={title} content={content}>
            <button>{t}</button>
          </Popover>
        ))}
      </>,
    );

    placements.forEach(t => {
      expect(getByText(t)).toMatchSnapshot();
    });
  });
});