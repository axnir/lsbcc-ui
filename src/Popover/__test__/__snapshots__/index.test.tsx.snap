// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popover /> should render default 1`] = `
<div>
  <div
    class="lsbcc-popover"
  >
    <div
      class="popover-container left hide"
    >
      <div
        class="popover-main"
      >
        <div
          class="popover-title"
        >
          Title
        </div>
        <div
          class="popover-content"
        >
          <div>
            <p>
              Hover me
            </p>
            <p>
              Hover me
            </p>
          </div>
        </div>
      </div>
      <div
        class="arrow left"
      />
    </div>
    <button
      class="popover-children"
    >
      left
    </button>
  </div>
</div>
`;

exports[`<Popover /> should render popover with placement 1`] = `
<button
  class="popover-children"
>
  top
</button>
`;

exports[`<Popover /> should render popover with placement 2`] = `
<button
  class="popover-children"
>
  right
</button>
`;

exports[`<Popover /> should render popover with placement 3`] = `
<button
  class="popover-children"
>
  bottom
</button>
`;

exports[`<Popover /> should render popover with placement 4`] = `
<button
  class="popover-children"
>
  left
</button>
`;
